### PR TITLE
Refactor normalize string method

### PR DESCRIPTION
**Pull request by jgwerner**
- Change `normalize_name_for_containers` to `normalize_string`
- Move normalize string tasks (username and course id) to setup_course_hook
- Update tests

#198

### CODE REVIEW

File: src/tests/illumidesk/authenticators/test_setup_course_hook.py [hunk 1]
```diff
@@ -53,6 +54,28 @@ async def test_setup_course_hook_raises_environment_error_with_missing_org(monke
...
 
 
+@pytest.mark.asyncio()
+async def test_setup_course_hook_calls_normalize_strings(
+    auth_state_dict, setup_course_environ, setup_course_hook_environ
+):
+    """
+    Does the setup_course_hook return normalized strings for the username and the course_id?
+    """
+    local_utils = LTIUtils()
```
Comment (netoisc) [inline src/tests/illumidesk/authenticators/test_setup_course_hook.py:64]: local_utils var seems not to be used

File: src/tests/illumidesk/authenticators/test_lti13_authenticator.py [hunk 2]
```diff
@@ -117,8 +117,7 @@ async def test_authenticator_returns_username_in_auth_state_with_with_name(monke
     ):
         with patch.object(LTI13LaunchValidator, 'validate_launch_request', return_value=True):
             result = await authenticator.authenticate(request_handler, None)
-            monkeypatch.setitem(auth_state_dict, 'name', 'foo')
-
+            auth_state_dict['name'] = 'Foo'
             assert result['name'] == auth_state_dict['name']
```
Comment (netoisc) [inline src/tests/illumidesk/authenticators/test_lti13_authenticator.py:121]: In this case we know exactly the value that the method should return us so I think that we can change the **assert** step with a fixed value eg. result['name'] == 'foo'

File: src/tests/illumidesk/authenticators/test_lti13_authenticator.py [hunk 3]
```diff
@@ -134,8 +133,7 @@ async def test_authenticator_returns_username_in_auth_state_with_with_given_name
     ):
         with patch.object(LTI13LaunchValidator, 'validate_launch_request', return_value=True):
             result = await authenticator.authenticate(request_handler, None)
-            monkeypatch.setitem(auth_state_dict, 'name', 'foobar')
-
+            auth_state_dict['name'] = 'Foo Bar'
             assert result['name'] == auth_state_dict['name']
```
Comment (netoisc) [inline src/tests/illumidesk/authenticators/test_lti13_authenticator.py:137]: the same, can we change it to have a fixed expected value?

File: src/illumidesk/authenticators/authenticator.py [hunk 4]
```diff
@@ -172,8 +175,7 @@ async def authenticate(self, handler: BaseHandler, data: Dict[str, str] = None)
             # runs as a docker container we need to normalize the string so we can use it
             # as a container name.
             if 'context_label' in args and args['context_label'] is not None:
-                course_label = args['context_label']
-                course_id = lti_utils.normalize_name_for_containers(course_label)
+                course_id = args['context_label']
                 self.log.debug('Course context_label normalized to: %s' % course_id)
```
Comment (netoisc) [inline src/illumidesk/authenticators/authenticator.py:179]: ok, I understand that the use of the `setup_course_hook` is required in the existing Authenticator classes to use, but what will happen if a new class of authenticator never calls that hook?  the names could be wrong right? Why not return the **course**, **username** always in lower case (or normalized) from our authenticators and checking it too in the hook? 
what do you think?

File: src/tests/illumidesk/authenticators/test_setup_course_hook.py (hunk 1 above)
Comment (jgwerner) [reply src/tests/illumidesk/authenticators/test_setup_course_hook.py:64]: Right, removing.

File: src/tests/illumidesk/authenticators/test_lti13_authenticator.py (hunk 2 above)
Comment (jgwerner) [reply src/tests/illumidesk/authenticators/test_lti13_authenticator.py:121]: Yes, simpler.

File: src/tests/illumidesk/authenticators/test_lti13_authenticator.py (hunk 3 above)
Comment (jgwerner) [reply src/tests/illumidesk/authenticators/test_lti13_authenticator.py:137]: Yes. I guess I got carried away removing the unnecessary monkeypatches.

File: src/illumidesk/authenticators/authenticator.py (hunk 4 above)
Comment (jgwerner) [reply src/illumidesk/authenticators/authenticator.py:179]: We could, but the root cause for the bug in the first place was because we were calling the normalize_string function point-to-point from the different authenticators and missed one of the calls for the course label. So as we add more items (potentially) to the `auth_state` dict then it gets harder to maintain.

I think for our setup we can assume that all authenticators will inherit from our base spawner class.

Comment (netoisc) [reply src/illumidesk/authenticators/authenticator.py:179]: but i remember that the auth_state dict is used by other classes and flows. I'm not sure if the `name` with spaces or in upper case will affected other parts.. but in general this change seems good
